docs/examples/pooling_utilities.md

```diff
@@ -0,0 +1,96 @@
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+mystnb:
+  execution_timeout: 100
+---
+# Pooling Utility
+
+Psignifit comes with an automatic pooling utility, which can pool trials of very similar stimulus levels into blocks to reduce the time required for fitting, to profit from the beta-binomial model and to improve interpretability of data-plots.
+
+In contrast to the matlab implementation, python-psignifit does not pool implicitly. Instead a warning is printed, if pooling might be useful. Then pooling can be run as a separate call using psignifit.tools.pool_blocks
+
+This way of pooling data is presumably acceptable for many datasets, but pooling by hand is probably more exact.
+
+```{code-cell} ipython3
+
+#act = pool_blocks(inp, max_tol=0.4, max_gap=10, max_length=10)
+```
+
+# Example
+For illustration we will use a dataset obtained from running a Quest run with 400 trials.
+
+```{code-cell} ipython3
+import numpy as np
+import psignifit as ps
+from psignifit import psigniplot
+from matplotlib import pyplot as plt
+
+data = np.genfromtxt("pooling_example_data.csv", delimiter=',', usecols=(0,1,2))
+print(data)
+```
+
+First we fit this dataset using the settings for a 2AFC experiment and a normal distribution.
+
+
+```{code-cell} ipython3
+options = {'sigmoid': 'norm',
+           'experiment_type': '2AFC'
+           }
+
+res = ps.psignifit(data, **options)
+```
+Note that this takes a bit longer than usual and it gives us a warning about the number per bloc. Let's have a look at what psignifit did automatically here:
+
+```{code-cell} ipython3
+fig, ax = plt.subplots(1, 1);
+psigniplot.plot_psychometric_function(res, ax=ax);
+```
+Each block contains only very few trials. Thus the beta-binomial model cannot help much to correct for overdispersion. Furthermore the many lines of data slow psignifit down.
+
+
+By default psignifit pools only trials collected at the exact same stimulus level. We can pool the data manually to make each bloc contain all trials which differ by up to 0.01.
+
+
+```{code-cell} ipython3
+from psignifit.tools import pool_blocks
+pooled = pool_blocks(data, max_tol=0.01)
+print(pooled)
+res = ps.psignifit(pooled, **options)
+fig, ax = plt.subplots(1, 1);
+psigniplot.plot_psychometric_function(res, ax=ax);
+```
+
+Now we pooled quite strongly.
+
+The other two options allow us to restrict which trials should be pooled again. For example we could restrict the number of trials to 25 per block:
+
+```{code-cell} ipython3
+pooled = pool_blocks(data, max_tol=0.01, max_length=25)
+print(pooled)
+res = ps.psignifit(pooled, **options)
+fig, ax = plt.subplots(1, 1);
+psigniplot.plot_psychometric_function(res, ax=ax);
+```
+
+This brakes the large blocks up again allowing us to notice if there was more variability over time than expected.
+
+The last option gives us a different rule to achieve something in a similar direction: We can enforce that a block must be collected en bloc like this:
```

Comment:
I'm not sure I got this: a block can be of unlimited size, and there is a maximum gap of 0 between blocks? Is it not like the first case without options?

Reply:
The first one pools all trials within a tolerance. The next one pools only up to a block size of 25. The last one pools only subsequent trials. If I understand correctly.

Reply:
got it, thanks. maybe "we can enforce to pool only subsequent trials", instead of "must be collected en block"?

Reply:
ok